apps/meteor/server/models/raw/LivechatInquiry.ts

```diff
@@ -91,4 +92,47 @@ export class LivechatInquiryRaw extends BaseRaw<ILivechatInquiryRecord> implemen
 			{ $unset: { locked: 1, lockedAt: 1 } },
 		);
 	}
+
+	async getSortingQuery(sort: Record<string, 1 | -1>): Promise<Document[]> {
+		const sortMechanism = await Settings.findOneById('Omnichannel_sorting_mechanism');
+		const $sort: { 'R.priorityWeight'?: number; 'estimatedServiceTimeAt'?: number; 'ts'?: number } = {};
+		const filter = [];
+		switch (sortMechanism?.value) {
+			case 'Priority':
+				$sort['R.priorityWeight'] = -1;
+				filter.push(
+					...[
+						{
+							$lookup: {
+								from: 'rocketchat_room',
+								localField: 'rid',
+								foreignField: '_id',
+								as: 'R',
+								pipeline: [
+									{
+										$project: {
+											priorityId: 1,
+											priorityWeight: 1,
+											_id: 0,
+										},
+									},
+								],
+							},
+						},
+						{ $unwind: '$R' },
+					],
```

Comment:
I like the idea of using an aggregation for this 🤔 but, I'm also afraid of the perf of this query/use under heavy load. Would be easier to attach the priorityId/weight to the inquiry for sorting. WDYT?

Reply:
We do the same for SLAs, that's why you can sort by estimatedServiceTimeAt without a lookup

Reply:
damn, should've though of doing that, it would've saved me much pain, and be faster at the same time.